### PR TITLE
Add release notes for 7.12.0

### PR DESCRIPTION
**Pull request by williamrandolph**
Release notes for 7.12 release.

### CODE REVIEW

File: docs/reference/release-notes/7.12.asciidoc [hunk 1]
```diff
@@ -0,0 +1,298 @@
+:es-issue: https://github.com/elastic/elasticsearch/issues/
+:es-pull:  https://github.com/elastic/elasticsearch/pull/
+
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:3]: These are already pulled in via our [docs attributes file](https://github.com/elastic/docs/blob/master/shared/attributes.asciidoc#L366). We can remove them here.
```suggestion
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+[[breaking-7.12.0]]
+[float]
+=== Breaking changes
+
+Mapping::
+* Returning nested data in fields API {es-pull}67432[#67432] (issue: {es-issue}63709[#63709])
+
+Query Languages::
+* "fields" api implementation in QL {es-pull}68802[#68802] (issue: {es-issue}67727[#67727])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:19]: Minor nit for consistency with previous bullet.
```suggestion
* Fields API implementation in QL {es-pull}68802[#68802] (issue: {es-issue}67727[#67727])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+SQL::
+* Test and fix the NULL handling of the String functions {es-pull}68379[#68379] (issue: {es-issue}58907[#58907])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:22]: Capitalization nit
```suggestion
* Test and fix the NULL handling of the string functions {es-pull}68379[#68379] (issue: {es-issue}58907[#58907])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+
+
+[[breaking-java-7.12.0]]
+[float]
+=== Breaking Java changes
+
+Features/Java High Level REST Client::
+* Move find file structure finder in Rest high Level client to its new endpoint and plugin {es-pull}67290[#67290]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:31]: Capitalization nit
```suggestion
* Move find file structure finder in REST High Level client to its new endpoint and plugin {es-pull}67290[#67290]
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+* Add deprecation warning for removed `strict_duplicate_detection` setting {es-pull}63500[#63500] (issues: {es-issue}34588[#34588], {es-issue}60222[#60222])
+
+Geo::
+* Deprecate `GeoPolygon` query in favour of `GeoShape` query {es-pull}64227[#64227] (issue: {es-issue}48928[#48928])
+
+Machine Learning::
+* Add deprecation check for job model snapshots that need upgraded {es-pull}66062[#66062] (issue: {es-issue}64154[#64154])
+
+Security::
+* Deprecate the id field for the `InvalidateApiKey` API {es-pull}66317[#66317] (issue: {es-issue}63224[#63224])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:52]:
```suggestion
* Deprecate the `id` field for the `InvalidateApiKey` API {es-pull}66317[#66317] (issue: {es-issue}63224[#63224])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+* Add a cluster privilege to cancel tasks and delete async searches {es-pull}68679[#68679] (issue: {es-issue}67965[#67965])
+
+EQL::
+* Introduce ~ grammar for case-insensitive functions {es-pull}67869[#67869] (issue: {es-issue}67868[#67868])
+* Introduce `like` and `regex` keywords {es-pull}68791[#68791] (issue: {es-issue}68639[#68639])
+
+Features/ILM+SLM::
+* Add support for partial searchable snapshots to ILM {es-pull}68714[#68714] (issues: {es-issue}68509[#68509], {es-issue}68605[#68605])
+* Add the frozen tier node role and ILM phase {es-pull}68605[#68605] (issue: {es-issue}60848[#60848])
+* Support `max_single_primary_size` in Resize Action and expose in ILM {es-pull}67705[#67705] (issue: {es-issue}65714[#65714])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:70]: Capitalization nit
```suggestion
* Support `max_single_primary_size` in resize action and expose in ILM {es-pull}67705[#67705] (issue: {es-issue}65714[#65714])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+
+Features/Ingest::
+* Fingerprint ingest processor {es-pull}68415[#68415] (issue: {es-issue}53578[#53578])
+* Network `community_id` processor for ingest pipelines {es-pull}66534[#66534] (issue: {es-issue}55685[#55685])
+* Network direction processor {es-pull}66644[#66644]
+
+Query Languages::
+* Implement support for `date_nanos` ES type {es-pull}67666[#67666] (issue: {es-issue}38562[#38562])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:78]:
```suggestion
* Implement support for `date_nanos` {es} field type {es-pull}67666[#67666] (issue: {es-issue}38562[#38562])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+SQL::
+* Implement TO_CHAR() function {es-pull}66486[#66486] (issue: {es-issue}54964[#54964])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:81]:
```suggestion
* Implement `TO_CHAR()` function {es-pull}66486[#66486] (issue: {es-issue}54964[#54964])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+Search::
+* Add minimum compatibility version to `SearchRequest` {es-pull}65896[#65896] (issue: {es-issue}63304[#63304])
+* Sort field tiebreaker for PIT (point in time) readers {es-pull}66093[#66093] (issue: {es-issue}56828[#56828])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:85]:
```suggestion
* Sort field tiebreaker for point in time (PIT) readers {es-pull}66093[#66093] (issue: {es-issue}56828[#56828])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+* `view_index_metadata` and `manage` privs now grant access to field capabilities {es-pull}67392[#67392] (issue: {es-issue}66867[#66867])
+
+Cluster Coordination::
+* Clarify message emitted on cluster UUID mismatch {es-pull}66915[#66915]
+* Extend default probe connect/handshake timeouts {es-pull}68059[#68059] (issue: {es-issue}42636[#42636])
+* Skip cluster state serialization to closed channel {es-pull}67413[#67413]
+
+EQL::
+* Add `result_position` to correctness test {es-pull}66761[#66761]
+* Introduce eql search status API {es-pull}68065[#68065] (issue: {es-issue}66955[#66955])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:121]:
```suggestion
* Introduce EQL search status API {es-pull}68065[#68065] (issue: {es-issue}66955[#66955])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+Infra/Plugins::
+* Notify users to restart Elasticsearch after plugin installation {es-pull}66723[#66723] (issue: {es-issue}39487[#39487])
+
+Infra/Scripting::
+* Add OSS whitelist to execute API {es-pull}67038[#67038] (issue: {es-issue}67035[#67035])
+* Capture structured javadoc from stdlib {es-pull}68782[#68782]
+* Enforce GPLv2 for parsed stdlib docs {es-pull}68601[#68601]
+* Improve bad regex pattern syntax error {es-pull}68520[#68520]
+* Parse stdlib files for parameter names {es-pull}67837[#67837]
+* Whitelist Json functions for ingest {es-pull}67118[#67118]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:173]:
```suggestion
* Whitelist JSON functions for ingest {es-pull}67118[#67118]
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+* Shortcut to avoid fmod {es-pull}66909[#66909]
+
+Packaging::
+* Tweaks to the Iron Bank docker context {es-pull}66942[#66942]
+
+Performance::
+* Add benchmark racing scripts {es-pull}68369[#68369]
+
+Query Languages::
+* Adapt nested fields extraction from "fields" API output to the new un-flattened structure {es-pull}68745[#68745] (issue: {es-issue}68722[#68722])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:202]:
```suggestion
* Adapt nested fields extraction from fields API output to the new un-flattened structure {es-pull}68745[#68745] (issue: {es-issue}68722[#68722])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+
+Engine::
+* Fail force-merges on read-only engines {es-pull}64756[#64756]
+
+Features/Features::
+* Prevent unnecessary system index access warnings in Deprecation Info API {es-pull}67312[#67312] (issue: {es-issue}66063[#66063])
+
+Features/ILM+SLM::
+* Avoid exception in in `SetStepInfoUpdateTask#onFailure` {es-pull}67553[#67553]
+* Fix Issues with CS Handling in ILM Async Steps {es-pull}68361[#68361]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.12.asciidoc:266]:
```suggestion
* Fix Issues with CS handling in ILM async steps {es-pull}68361[#68361]
```

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks for raising this @williamrandolph!

I left a few minor nits but nothing blocking.

I'm also going to add the @elastic/ml-docs team as reviewers so they can review their sections.

**Comment by williamrandolph** [comment]
@jrodewig Thank you for the review! I've incorporated your suggestions.

**Review by lcawl — APPROVED** [review]
I've added ml-cpp PRs from https://github.com/elastic/ml-cpp/blob/7.12/docs/CHANGELOG.asciidoc. Content LGTM now

**Comment by jrodewig** [comment]
I'm going to merge this so the RN link is available. I'll open a separate PR to remove the coming tags after release. Thanks again @williamrandolph!

**Pull request merged**